src/test/java/io/cdap/plugin/gcp/dataplex/sink/config/DataplexBatchSinkConfigTest.java

```diff
@@ -0,0 +1,651 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.dataplex.sink.config;
+
+import com.google.api.gax.rpc.ApiException;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.dataplex.v1.Asset;
+import com.google.cloud.dataplex.v1.AssetName;
+import com.google.cloud.dataplex.v1.DataplexServiceClient;
+import com.google.cloud.dataplex.v1.Lake;
+import com.google.cloud.dataplex.v1.LakeName;
+import com.google.cloud.dataplex.v1.Zone;
+import com.google.cloud.dataplex.v1.ZoneName;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import io.cdap.plugin.gcp.common.GCPConnectorConfig;
+import io.cdap.plugin.gcp.dataplex.common.config.DataplexBaseConfig;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.internal.util.reflection.FieldSetter;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({DataplexBatchSinkConfig.class, Schema.class, DataplexServiceClient.class})
+public class DataplexBatchSinkConfigTest {
+
+  @Mock
+  StatusCode statusCode;
+
+  @Test
+  public void validateBigQueryDatasetTest() throws NoSuchFieldException {
+    DataplexBatchSinkConfig dataplexBatchSinkConfig = new DataplexBatchSinkConfig();
+    MockFailureCollector mockFailureCollector = new MockFailureCollector();
+    FieldSetter.setField(dataplexBatchSinkConfig, DataplexBatchSinkConfig.class.getDeclaredField("table"), "test");
```

Comment:
Similar comment, instead of `FieldSetter` can we use builder with Private Constructor in all the tests. This makes the tests brittle if we modified the field name in future.

Reply:
done